### DISCUSSION TITLE
How to understand C++ codes easily

### DISCUSSION THREAD

**Discussion by usstq** [usstq/openvino#14] [General]
C++ codes are hard to understand because of the complex syntax, how to understand it easily w/o need a lot of readings?

**Comment by usstq** [comment]
Simplify the code you are trying to understand and put into godbolt.org:

https://godbolt.org/

for example, [this piece of code](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:2,endLineNumber:21,positionColumn:1,positionLineNumber:1,selectionStartColumn:2,selectionStartLineNumber:21,startColumn:1,startLineNumber:1),source:'//+Type+your+code+here,+or+load+an+example.%0A%0Astruct+shape%7B%0A++++int+a%3B%0A++++virtual+void+size()%7B%7D%0A%7D%3B%0A%0Astruct+rect:+public+shape%7B%0A++++int+b%3B%0A++++void+size()+override%7B%7D%0A%7D%3B%0A%0A%0Avoid+test()%7B%0A%0A++++rect+s%3B%0A++++s.size()%3B%0A%0A++++shape+*+p0+%3D+%26s%3B%0A++++p0-%3Esize()%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:g112,filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'',selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1,tree:'1'),l:'5',n:'0',o:'x86-64+gcc+11.2+(C%2B%2B,+Editor+%231,+Compiler+%231)',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)  demonstrated the object memory layout, virtual table data structure:
 - vtable for rect
 - vtable for shape
 - how vptr is initialized in `shape::shape()` and `rect::rect()`
 - how type down cast the done

**Reply by yayaisdanger** [reply]
:)
